x/oracle/keeper/nonce.go

```diff
@@ -0,0 +1,175 @@
+package keeper
+
+import (
+	"errors"
+
+	"github.com/ExocoreNetwork/exocore/x/oracle/keeper/common"
+	"github.com/ExocoreNetwork/exocore/x/oracle/types"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// GetNonce get the nonce for a specific validator
+func (k Keeper) GetNonce(ctx sdk.Context, validator string) (nonce types.ValidatorNonce, found bool) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.NonceKeyPrefix))
+	return k.getNonce(store, validator)
+}
+
+// SetNonce set the nonce for a specific validator
+func (k Keeper) SetNonce(ctx sdk.Context, nonce types.ValidatorNonce) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.NonceKeyPrefix))
+	k.setNonce(store, nonce)
+}
+
+// AddNonceItem add a nonce item for a specific validator
+func (k Keeper) AddNonceItem(ctx sdk.Context, nonce types.ValidatorNonce) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.NonceKeyPrefix))
+	if n, found := k.getNonce(store, nonce.Validator); found {
+		feederIDs := make(map[uint64]struct{})
+		for _, v := range n.NonceList {
+			feederIDs[v.FeederID] = struct{}{}
+		}
+		for _, v := range nonce.NonceList {
+			if _, ok := feederIDs[v.FeederID]; ok {
+				continue
+			}
+			n.NonceList = append(n.NonceList, v)
+		}
+		k.setNonce(store, n)
+	} else {
+		k.setNonce(store, nonce)
+	}
+}
+
+// AddZeroNonceItemForValidators init the nonce of a specific feederID for a set of validators
+func (k Keeper) AddZeroNonceItemWithFeederIDForValidators(ctx sdk.Context, feederID uint64, valdiators []string) {
```

Comment:
valdiators : typo 
can check validators firstly.

>     if len(validators) == 0 {
        return 
    }

Reply:
This is used to init for dogfood validators, and that validator set should not be empty